gap/display.gi

```diff
@@ -68,19 +84,53 @@ function(D)
   Append(str, "graph hgn{\n");
   Append(str, "node [shape=circle]\n\n");
   for i in DigraphVertices(D) do
-    Append(str, Concatenation(String(i), "\n"));
+    # Append(str, StringFormatted("{}", i));
+    Append(str, String(i));
+    for func in node_funcs do
+      Append(str, func(i));
+    od;
+    Append(str, "\n");
   od;
   for i in DigraphVertices(D) do
     for j in out[i] do
       if j >= i then
-        Append(str, Concatenation(String(i), " -- ", String(j), "\n"));
+        # Append(str, StringFormatted("{} -- {}", i, j));
+        Append(str, String(i));
+        Append(str, " -- ");
+        Append(str, String(j));
+        for func in edge_funcs do
+          Append(str, func(i, j));
+        od;
+        Append(str, "\n");
       fi;
     od;
   od;
   Append(str, "}\n");
   return str;
 end);
 
+InstallMethod(DotSymmetricDigraph, "for a digraph by out-neighbours",
+[IsDigraphByOutNeighboursRep],
+function(D)
+  return DIGRAPHS_DotSymmetricDigraph(D, [], []);
+end);
+
+InstallMethod(DotSymmetricColoredDigraph,
+"for a digraph by out-neighbours and two lists",
+[IsDigraphByOutNeighboursRep, IsList, IsList],
+function(D, vert, edge)
+  return DIGRAPHS_DotSymmetricDigraph(D,
```

Comment:
What happens if the list of colors (edge or vertex) is not valid? I.e. if the list is too short? Or has holes?